### PR TITLE
[PRM-643] Update CYA Page with filenames

### PR DESCRIPTION
**Pull request by JBlacklaws**

### CODE REVIEW

File: app/controllers/CheckYourAnswersController.scala [hunk 1]
```diff
@@ -16,54 +16,79 @@
...
+              val previousUploadsFileName = previousUploads.map( _.map( file => file.uploadDetails.map( details => details.fileName)))
+              val answersFromSession = sessionAnswersHelper.getAllTheContentForCheckYourAnswersPage(Some(previousUploadsFileName.getOrElse(Seq.empty).collect{
+                case Some(x) => x
+              }.mkString(", ")))
+              Ok(checkYourAnswersPage(answersFromSession))
+            }
+          } else {
+            val answersFromSession = sessionAnswersHelper.getAllTheContentForCheckYourAnswersPage()
+            Future(Ok(checkYourAnswersPage(answersFromSession)))
+          }
```
Comment (LewisMitchell) [inline app/controllers/CheckYourAnswersController.scala:86]: You could replace this check with this:
```
            logger.debug(s"[CheckYourAnswersController][onPageLoad] Loading check your answers page")
            for {
              previousUploads <- uploadJourneyRepository.getUploadsForJourney(request.session.get(SessionKeys.journeyId))
            } yield {
              val previousUploadsFileName = previousUploads.map(_.map(file => file.uploadDetails.map(details => details.fileName)))
              val x = previousUploadsFileName.getOrElse(Seq.empty).collect {
                case Some(x) => x
              }.mkString(", ")
              val answersFromSession = sessionAnswersHelper.getAllTheContentForCheckYourAnswersPage(if (x.isEmpty) None else Some(x))
              Ok(checkYourAnswersPage(answersFromSession))
            }
```

Comment (LewisMitchell) [reply app/controllers/CheckYourAnswersController.scala:86]: This way saves the 2 different scenarios where the parameter is Some(x) or None

**Review by LewisMitchell — CHANGES_REQUESTED** [review]
AT's fail, most likely because my https://github.com/hmrc/penalties-appeals-frontend/pull/147 PR isn't in.
Just some changes to refactor the implementation a bit more

File: app/controllers/CheckYourAnswersController.scala [hunk 2]
```diff
@@ -70,19 +71,15 @@ class CheckYourAnswersController @Inject()(checkYourAnswersPage: CheckYourAnswer
...
-              Ok(checkYourAnswersPage(answersFromSession))
-            }
-          } else {
-            val answersFromSession = sessionAnswersHelper.getAllTheContentForCheckYourAnswersPage()
-            Future(Ok(checkYourAnswersPage(answersFromSession)))
+          for {
+            previousUploads <- uploadJourneyRepository.getUploadsForJourney(request.session.get(SessionKeys.journeyId))
+          } yield {
+            val previousUploadsFileName = previousUploads.map(_.map(file => file.uploadDetails.map(details => details.fileName)))
+            val x = previousUploadsFileName.getOrElse(Seq.empty).collect {
```
Comment (LewisMitchell) [inline app/controllers/CheckYourAnswersController.scala:78]: This is my mistake :sorry: . But can you make `x` to be something like `fileNames`?

File: app/controllers/CheckYourAnswersController.scala [hunk 3]
```diff
@@ -56,10 +56,11 @@ class CheckYourAnswersController @Inject()(checkYourAnswersPage: CheckYourAnswer
           for {
             previousUploads <- uploadJourneyRepository.getUploadsForJourney(request.session.get(SessionKeys.journeyId))
           } yield {
-            val previousUploadsFileName = previousUploads.map( _.map( file => file.uploadDetails.map( details => details.fileName)))
-            val answersFromSession = sessionAnswersHelper.getAllTheContentForCheckYourAnswersPage(Some(previousUploadsFileName.getOrElse(Seq.empty).collect{
+            val previousUploadsFileName = previousUploads.map(_.map(file => file.uploadDetails.map(details => details.fileName)))
+            val x = previousUploadsFileName.getOrElse(Seq.empty).collect {
```
Comment (LewisMitchell) [inline app/controllers/CheckYourAnswersController.scala:60]: This is my mistake :sorry: . But can you make `x` to be something like `fileNames`?

**Review by LewisMitchell — CHANGES_REQUESTED** [review]

**Comment by LewisMitchell** [comment]
Also can we remove evidenceFileName now? We shouldn't need it anymore?

**Comment by JBlacklaws** [comment]
Yeah, we should get rid of that SessionKey, but that's uncovered a bunch of TODOs I should have handled, so I'll need to clear those out!

**Review by LewisMitchell — CHANGES_REQUESTED** [review]
This also needs an AT change, something either temporary or more permanent.
See AppealAgainstObligation flow - Changing Values scenario for more details.
It uploads another file and the CYA page displays the old file and new file (something we will remove later - but needs to be addressed in the AT's now)

**Review by LewisMitchell — APPROVED** [review]

**Comment by LewisMitchell** [comment]
AT change https://github.com/hmrc/penalties-acceptance-tests/pull/80

**Pull request merged**